### DISCUSSION TITLE
Visual Studio 2019  Windows SDK version 10.0.22621.0 was not found

### DISCUSSION THREAD

**Discussion by cgdestroyer** [AcademySoftwareFoundation/OpenRV#75] [Build Issues]
Got this error  Microsoft Visual Studio\2019  - The Windows SDK version 10.0.22621.0 was not found. 

C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.Cpp.WindowsSDK.targets(46,5): error MSB8036: The Windows SDK version 10.0.22621.0 was not found. 

https://i.imgur.com/SXoT0wJ.jpg

but there is no such version in VS 2019 but there is for 2022 but having VS 2022 case another problems, seems like every day there are differnet sets of problems here  :) 
https://i.imgur.com/SBKQ1pt.jpg

**Comment by p2or** [comment]
Hi @cgdestroyer,

just to share my experience: I had no issues while compiling yesterday using 'VS2019 Community Edition' and `10.0.190041.0` SDK as well as 'VS2017 Community Edition' with the latest SDK it supports (`10.0.17763.0`).

Hope this helps,
Christian

**Comment by rogernelson** [comment]
Hi @cgdestroyer there are definitely issues building with VS2022 installed, as some of the third-party dependencies automatically detect the latest version of MSVC and build with it, even if they do not support it.  I haven't been able to find a workaround to that yet.

**Comment by cgdestroyer** [comment]
@rogernelson this error was after I unistall VS 2022. I even tried vierutal machine with only nessery items installled and still had errors :/

**Comment by cgdestroyer** [comment]
This is warning and error I'm getting on virtual machine, win 10 64 and istalled only wnat is needed

https://i.imgur.com/17iXyh0.jpg
https://i.imgur.com/pzFoPN1.jpg

**Comment by rogernelson** [comment]
The first one is just a warning, it shouldn't prevent the build from completing.  For the second one, PR https://github.com/AcademySoftwareFoundation/OpenRV/pull/31 will likely fix it.  If you pull/rebase on the latest `main` branch do you still have the problem?

**Comment by cgdestroyer** [comment]
@rogernelson So I shouidl just redownload rv from github and it shoudl work?
I deleted rvopen folder form C drive, then used this 
cd /c
git clone https://github.com/AcademySoftwareFoundation/OpenRV.git

then
cd /c/openrv
cmake -B cmake-build -G "Visual Studio 16 2019" -A x64 -DCMAKE_BUILD_TYPE=Release -DRV_DEPS_WIN_PERL_ROOT=/c/Strawberry/perl/bin -DRV_DEPS_QT5_LOCATION=/c/Qt/5.15.2/msvc2019_64
cmake --build cmake-build --config Release --target rv -j8 
cmake --install cmake-build --config Release

and got this error so something is still wrong 
https://i.imgur.com/OAzO3z4.jpg

**Comment by rogernelson** [comment]
Hi @cgdestroyer.  There's no need to delete and redownload, simply doing a `git pull` is enough (that will only get and build the changes since your last pull/clone).  

You have a different error now, so it seems your previous one is solved by the update.  Although, I have never run into that aclocal error myself, another user did was able to solve it here: https://github.com/AcademySoftwareFoundation/OpenRV/issues/3#issuecomment-1397634886
